api.go

```diff
@@ -27,10 +27,10 @@ type API struct {
 	FilterStructMemberGetter func(m *StructMember) bool
 }
 
-func (a *API) HandleDirectory(dir string) error {
+func (a *API) HandleDirectory(dir string) (*HeaderFile, error) {
```

Comment:
The HeaderFile is internally misused in this method. The same object is used for multiple times and the name is adapted for every file. This PR is a good chance to clean this up. I think we should distinguish between HeaderFile and Directory(or some other name). At the moment the HeaderFile.handle method handles ALL header files of one directory.

Sidenote: Please try to put API changes into their own commits.